src/pages/court-cases/[courtCaseId]/submit.tsx

```diff
@@ -0,0 +1,181 @@
+      previousPath: previousPath ?? null,
+      user: userToDisplayFullUserDto(currentUser),
+      courtCase: courtCaseToDisplayFullCourtCaseDto(courtCase),
+      hasAmendments: false,
+      amendments: "{}"
+    }
+
+    if (isPost(req)) {
+      const { amendments } = formData as { amendments: string }
+
+      const parsedAmendments = JSON.parse(amendments)
+
+      const hasAmendments = Object.keys(parsedAmendments).length > 0
+      props.hasAmendments = hasAmendments
+      props.amendments = amendments
+
+      return { props }
+    } else {
+      redirectTo(previousPath ?? `/court-cases/${courtCase.errorId}`)
+    }
+
+    if (isPost(req) && confirm) {
+      const { amendments } = formData as { amendments: string }
+
+      const parsedAmendments = JSON.parse(amendments)
+
+      const updatedAmendments =
+        Object.keys(parsedAmendments).length > 0 ? parsedAmendments : { noUpdatesResubmit: true }
+
+      const amendedCase = await resubmitCourtCase(dataSource, updatedAmendments, +courtCaseId, currentUser)
+
+      if (isError(amendedCase)) {
+        throw amendedCase
+      }
+
+      let redirectUrl = `/court-cases/${courtCase.errorId}`
+
+      if (previousPath) {
+        redirectUrl += `?previousPath=${encodeURIComponent(previousPath)}`
+      }
+
+      if (!currentUser.hasAccessTo[Permission.Triggers] || courtCase.triggerStatus !== "Unresolved") {
+        return redirectTo("/")
+      }
+      return redirectTo(redirectUrl)
+    }
+    return { props }
+  }
+)
+interface Props {
+  user: DisplayFullUser
+  courtCase: DisplayFullCourtCase
+  csrfToken: string
+  previousPath: string | null
+  hasAmendments?: boolean
+  amendments?: string
+}
+const SubmitCourtCasePage: NextPage<Props> = ({
+  courtCase,
+  user,
+  previousPath,
+  hasAmendments = false,
+  amendments = "{}",
+  csrfToken
+}: Props) => {
+  const { basePath } = useRouter()
+  const [currentUserContext] = useState<CurrentUserContextType>({ currentUser: user })
+  let backLink = `${basePath}/court-cases/${courtCase.errorId}`
+  if (previousPath) {
+    backLink += `?previousPath=${encodeURIComponent(previousPath)}`
+  }
+  const resubmitCasePath = `${basePath}/court-cases/${courtCase.errorId}?resubmitCase=true`
+  return (
+    <>
+      <CurrentUserContext.Provider value={currentUserContext}>
+        <Layout>
+          <Head>
+            <title>{"Submit Case Exception(s) | Bichard7"}</title>
+            <meta name="description" content="Submit Case Exception(s) | Bichard7" />
+          </Head>
+          <BackLink href={backLink} onClick={function noRefCheck() {}}>
+            {"Case Details"}
+          </BackLink>
+          <HeaderContainer id="header-container">
+            <HeaderRow>
+              <Heading as="h1" size="LARGE" aria-label="Submit Exception(s)">
+                {"Submit Exception(s)"}
+              </Heading>
+            </HeaderRow>
+          </HeaderContainer>
+
+          <ConditionalRender isRendered={hasAmendments}>
```

Comment:
Could use 
```suggestion
          <ConditionalRender isRendered={amendments.length > 0}>
```
so `hasAmendments` is not needed

Reply:
The reason `hasAmendments` is needed is because amendments is initialized `amendments: "{}"`. So `amendments.length` will always return 2. I would be happy to discuss alternative approaches.